File: app/src/main/java/br/com/onibusapp/onibusapp/MainActivity.java
package br.com.onibusapp.onibusapp;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import br.com.onibusapp.onibusapp.data.ScriptDB;
import br.com.onibusapp.onibusapp.ui.MapsFragment;
import br.com.onibusapp.onibusapp.ui.favoritos.FavoritosFragment;
import br.com.onibusapp.onibusapp.ui.pesquisa.PesquisarFragment;
import br.com.onibusapp.onibusapp.utils.FragmentUtil;

public class MainActivity extends AppCompatActivity {

    private FragmentUtil fragmentUtil;
    private ScriptDB scriptDB;

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    fragmentUtil.mudarTela(new MapsFragment());
                    return true;
                case R.id.navigation_dashboard:
                    fragmentUtil.mudarTela(new PesquisarFragment());
                    return true;
                case R.id.navigation_notifications:
                    fragmentUtil.mudarTela(new FavoritosFragment());
                    return true;
            }
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
        this.fragmentUtil = FragmentUtil.getInstance(getSupportFragmentManager());
        inicializaTelaMapa();
        scriptDB = new ScriptDB(this);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }

    private void inicializaTelaMapa() {
        fragmentUtil.mudarTela(new MapsFragment());
    }

}

File: app/src/main/java/br/com/onibusapp/onibusapp/MainActivity.kt
package br.com.onibusapp.onibusapp

import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v7.app.AppCompatActivity
import android.view.View
import br.com.onibusapp.onibusapp.data.ScriptDB
import br.com.onibusapp.onibusapp.ui.MapsFragment
import br.com.onibusapp.onibusapp.ui.favoritos.FavoritosFragment
import br.com.onibusapp.onibusapp.ui.pesquisa.PesquisarFragment
import br.com.onibusapp.onibusapp.utils.FragmentUtil
import com.google.firebase.FirebaseApp

class MainActivity : AppCompatActivity() {

    private var fragmentUtil: FragmentUtil? = null
    private var scriptDB: ScriptDB? = null


    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_home -> {
                fragmentUtil!!.mudarTela(MapsFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_dashboard -> {
                fragmentUtil!!.mudarTela(PesquisarFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_notifications -> {
                fragmentUtil!!.mudarTela(FavoritosFragment())
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val navigation = findViewById<View>(R.id.navigation) as BottomNavigationView
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        this.fragmentUtil = FragmentUtil.getInstance(supportFragmentManager)
        inicializaTelaMapa()
        scriptDB = ScriptDB(this)
        FirebaseApp.initializeApp(this)
    }

    override fun onBackPressed() {
        super.onBackPressed()
    }

    private fun inicializaTelaMapa() {
        fragmentUtil!!.mudarTela(MapsFragment())
    }

}
